js/index.js

```diff
@@ -1,9 +1,353 @@
+    document.getElementById("question-nickname").style.display = "";
+    document.getElementById("question-face").style.display = "none";
+    load(); //ローカルストレージ読込
+    displayScores(); //成績表示
+  }
+
+  //ニックネームが選択されたときの処理
+  const nickname = document.getElementById("nickname");
+  nickname.addEventListener('click', () => {
+    nicknameMode = true;
+    faceMode = false;
+    document.getElementById("question-nickname").style.display = "";
+    document.getElementById("question-face").style.display = "none";
+  });
+
+  //顔ボタンが選択されたときの処理
+  const face = document.getElementById("face");
+  face.addEventListener('click', () => {
+    nicknameMode = false;
+    faceMode = true;
+    document.getElementById("question-face").style.display = "";
+    document.getElementById("question-nickname").style.display = "none";
+  });
+
+  //ダイアログ関連
+  const dialog = document.getElementById("dialog");
+  const resultMessage = document.getElementById("result_message");
+  const nextButton = document.getElementById("nextButton");
+
+  nextButton.addEventListener("click", clickNextButton);
+
+  // 最終問題かどうかを判定する
+  function isQuestionEnd() {
+    return count === questionLength;
+  }
+
+  //結果表示処理
+  function setResult() {
+    // 正解率を計算する
+    accuracyRate = Math.floor((correctCount / questionLength) * 100);
+    // 正解率を表示する
+    resultMessage.innerText = `お疲れ様でした^^ \n あなたの正解数は、\n ${correctCount}問/${questionLength}問`;
+    resultMessage.style.fontSize = "24px";
+    if(accuracyRate >= 80){
+      result_shape.innerText = `おめでとうございます!!`;
+    }
+    if((accuracyRate < 80) && (accuracyRate >= 60)){
+      result_shape.innerText = `まずまずです！`;
+    }
+    if(accuracyRate < 60){
+      result_shape.innerText = `もう少し頑張りましょう。`;
+    }
+    nextButton.innerText = "閉じる";
+    resultMessage.style.color = " #ff7800";
+    result_shape.style.fontSize = "24px";
+    result_shape.style.color = " #ff7800";
+    dialog.showModal();
+  }
+
+  function displaydialog(result) {
+    // 正解かどうかを判定する
+    if (result === true) {
+      correctCount++;
+      resultMessage.innerText = "正解です!!";
+      resultMessage.style.color = "#de2956";
+      result_shape.innerText = "⭕️";
+    } else {
+      resultMessage.innerText = "不正解です!!";
+      resultMessage.style.color = "#785dc8";
+      result_shape.innerText = "✖️";
+    }
+    // 最後の問題かどうかを判定する
+    if (isQuestionEnd()) {
+      nextButton.innerText = "結果を見る";
+    } else {
+      nextButton.innerText = "次の問題へ";
+    }
+    // ダイアログを表示する
+    dialog.showModal();
+  }
+
+  function clickNextButton() {
+    if (isQuestionEnd()) {
+      //最終問題かつ結果表示までされているとき
+      if (nextButton.innerText == "閉じる") {
+        dialog.close();
+        save();  //ローカルストレージに保存
+        location.reload(); //リロード
+      }
+      else {
+        //アニメーションを一時停止
+        stopAnimation();
+        // クイズの結果画面に結果を設定する
+        setResult();
+      }
+    } else {
+      // クイズの問題画面に問題を設定する
+      setQuestion();
+      dialog.close();
+    };
```

Comment:
if文はネストしがちですが、可読性を落とさないためにあまりネストしないほうがいいです。
結果が分かるモノは早めにreturnするのが良いです。

```suggestion
    if (!isQuestionEnd()) {
      // クイズの問題画面に問題を設定する
      setQuestion();
      dialog.close();
      return
    }
    //最終問題かつ結果表示までされているとき
    if (nextButton.innerText == "閉じる") {
      dialog.close();
      save();  //ローカルストレージに保存
      location.reload(); //リロード
      return
    }
    //アニメーションを一時停止
    stopAnimation();
    // クイズの結果画面に結果を設定する
    setResult();
```